### ISSUE TITLE
Improvements needed for vapp and vapp_vm import

### ISSUE THREAD

**Issue by jpbuecken** [terraform-providers/terraform-provider-vcd#448]
As you mentioned in

https://github.com/terraform-providers/terraform-provider-vcd/issues/379#issue-514578890

the power_on value is not set during import.

But this has a big issue if you want to import. 
Assume you want to change something afterwards. Something simple that does not need a reboot.

BUT:

The next `terraform apply` will reboot your vApp and your VM, even if you ignore `power_on` in lifecycle state.

Reason:
The state for power_on is null after import. It looks like the ignore in lifecycle does not work for items that have a 'null' value.

Steps to reproduce
Example for vapp, same result for vm:
- Create a HCL that ignores power_on in lifecycle for vapp. (Optional)
- Import vapp
- Run terraform apply
- vApp (and VM) has been rebooted

HCL:
```hcl
resource "vcd_vapp" "vapp" {
  lifecycle {
    prevent_destroy = "true"
    ignore_changes = [
      power_on,
    ]
  }

  name = var.hostname
}
```

```
> terraform import vcd_vapp.vapp $VCD_ORG.$VCD_VDC.test1
vcd_vapp.vapp: Importing from ID "XXX.YYY.test1"...
vcd_vapp.vapp: Import prepared!
  Prepared vcd_vapp for import
vcd_vapp.vapp: Refreshing state... [id=urn:vcloud:vapp:0036a957-44ec-46c7-9324-448b5d1f0034]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

> grep power_on terraform.tfstate
            "power_on": null,

> terraform apply
vcd_vapp.vapp: Refreshing state... [id=urn:vcloud:vapp:0036a957-44ec-46c7-9324-448b5d1f0034]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # vcd_vapp.vapp will be updated in-place
  ~ resource "vcd_vapp" "vapp" {
      + accept_all_eulas = true
        guest_properties = {}
        href             = "https://XXX/api/vApp/vapp-0036a957-44ec-46c7-9324-448b5d1f0034"
        id               = "urn:vcloud:vapp:0036a957-44ec-46c7-9324-448b5d1f0034"
        ip               = "allocated"
        metadata         = {}
        name             = "test1"
        status           = 4
        status_text      = "POWERED_ON"
    }

```

**Comment by jpbuecken** [comment]
Maybe hardcode `true` or `"dummy"` as import value? 
At least ignore_changes will work for customer who don't care about the value and this is the default create value.
You can still change the power_on value in HCL and manage your power state.

Not sure why it is not reliable. Can't you get it form status value? 
e.g.
`status           = 4`
?

**Comment by Didainius** [comment]
True. Thanks for this validation. This needs some extra thinking and seeing what Terraform SDK can give us to handle it better.

**Comment by dataclouder** [comment]
> Not sure why read for power_on is not reliable. Can't you get it form status value?
> e.g.
> `status = 4`
> ?

Power status is not easy to define.
It should be set when status == 4, but even when we request it, the status change may not happen until late. If the status is, for example, "RESOLVED" (status code 1), we may report it as powered off, while it is still starting, and it will reach the POWER_ON state soon.

Returning the power status at an early stage may mislead both users and terraform, with the result of having unnecessary update requests.
We have two fields that produce better information:
* status (numeric status code, such as "4")
* status_text (status as a string, such as "POWERED_ON")
